From eac92d06bb2ec66ee557a1149f9d010165094c90 Mon Sep 17 00:00:00 2001
From: Daniel Kolesa <daniel@octaforge.org>
Date: Sat, 26 Mar 2022 04:00:49 +0100
Subject: [PATCH] meson: only include header file in the declared libosinfo dep

This prevents compiling a separate copy of the enums .c file
into anything using the libosinfo_dep target. Doing so would
result in types occasionally getting registered twice, at least
when libosinfo is built with Clang.
---
 osinfo/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osinfo/meson.build b/osinfo/meson.build
index cdd150a..a7c5a35 100644
--- a/osinfo/meson.build
+++ b/osinfo/meson.build
@@ -161,7 +161,7 @@ libosinfo = library(
 
 libosinfo_dep = declare_dependency(
     sources: [
-        libosinfo_enum_types,
+        libosinfo_enum_types[1],
     ],
     link_with: libosinfo,
     dependencies: libosinfo_dependencies,
-- 
2.35.1

